public/samples/DataStreams/StreamsUpkeep.sol

```diff
@@ -110,65 +98,48 @@ contract StreamsUpkeep is ILogAutomation, StreamsLookupCompatibleInterface {
         return (true, abi.encode(values, extraData));
     }
 
-    function performUpkeep(bytes calldata performData) external override {
+    // function will be performed on-chain
+    function performUpkeep(bytes calldata performData) external {
+        // Decode incoming performData
         (bytes[] memory signedReports, bytes memory extraData) = abi.decode(
             performData,
             (bytes[], bytes)
         );
 
         bytes memory report = signedReports[0];
 
-        bytes memory bundledReport = bundleReport(report);
+        (, bytes memory reportData) = abi.decode(report, (bytes32[3], bytes));
 
-        BasicReport memory unverifiedReport = _getReportData(report);
+        // Billing
+        IFeeManager feeManager = IFeeManager(address(verifier.s_feeManager()));
+        IRewardManager rewardManager = IRewardManager(
+            address(feeManager.i_rewardManager())
+        );
 
-        bytes memory verifiedReportData = verifier.verify{
-            value: unverifiedReport.nativeFee
-        }(bundledReport);
-        BasicReport memory verifiedReport = abi.decode(
-            verifiedReportData,
-            (BasicReport)
+        address feeTokenAddress = feeManager.i_linkAddress();
+        (Common.Asset memory fee, , ) = feeManager.getFeeAndReward(
+            address(this),
+            reportData,
+            feeTokenAddress
         );
 
-        emit PriceUpdate(verifiedReport.price);
-    }
+        // Approve fee spend
```

Comment:
May be helpful to expand a bit here and include an implicit reminder that the contract must be funded:
// Approve rewardManager to spend this contract's balance in fees.

Reply:
Done